fairlearn/metrics/group_metric_result.py

```diff
@@ -0,0 +1,17 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+
+class GroupMetricResult:
+    def __init__(self):
+        # The 'metric' property is the given metric evaluated without regard to group
```

Comment:
The object itself is a "metric result", so i'm not sure we need to repeat "metric" in each field. I'm in favor of the following replacements... please treat them as suggestions--only if they seem clear!!!
.metric -> .overall
.group_metrics -> .by_group
.min_metric -> .min (or: min_over_groups)
.max_metric -> .max (or: max_over_groups)
.min_metric_groups -> .argmin (to reflect numpy, or possibly: argmin_groups)
.max_metric_groups -> .argmax (to reflect numpy, or possibly: argmax_groups)
.metric_range -> .range (or: .disparity_difference)
.metric_range_ratio -> .range_ratio (or: .disparity_ratio)

Reply:
I've added _over_groups to a few, but changed